src/main/java/ch/njol/skript/expressions/arithmetic/ExprArithmetic.java

```diff
@@ -0,0 +1,200 @@
+		}
+	}
+	
+	private final static Patterns<PatternInfo> patterns = new Patterns<>(new Object[][] {
+
+		{"\\(%number%\\)[ ]+[ ]\\(%number%\\)", new PatternInfo(Operator.PLUS, true, true)},
+		{"\\(%number%\\)[ ]+[ ]%number%", new PatternInfo(Operator.PLUS, true, false)},
+		{"%number%[ ]+[ ]\\(%number%\\)", new PatternInfo(Operator.PLUS, false, true)},
+		{"%number%[ ]+[ ]%number%", new PatternInfo(Operator.PLUS, false, false)},
+		
+		{"\\(%number%\\)[ ]-[ ]\\(%number%\\)", new PatternInfo(Operator.MINUS, true, true)},
+		{"\\(%number%\\)[ ]-[ ]%number%", new PatternInfo(Operator.MINUS, true, false)},
+		{"%number%[ ]-[ ]\\(%number%\\)", new PatternInfo(Operator.MINUS, false, true)},
+		{"%number%[ ]-[ ]%number%", new PatternInfo(Operator.MINUS, false, false)},
+		
+		{"\\(%number%\\)[ ]*[ ]\\(%number%\\)", new PatternInfo(Operator.MULT, true, true)},
+		{"\\(%number%\\)[ ]*[ ]%number%", new PatternInfo(Operator.MULT, true, false)},
+		{"%number%[ ]*[ ]\\(%number%\\)", new PatternInfo(Operator.MULT, false, true)},
+		{"%number%[ ]*[ ]%number%", new PatternInfo(Operator.MULT, false, false)},
+		
+		{"\\(%number%\\)[ ]/[ ]\\(%number%\\)", new PatternInfo(Operator.DIV, true, true)},
+		{"\\(%number%\\)[ ]/[ ]%number%", new PatternInfo(Operator.DIV, true, false)},
+		{"%number%[ ]/[ ]\\(%number%\\)", new PatternInfo(Operator.DIV, false, true)},
+		{"%number%[ ]/[ ]%number%", new PatternInfo(Operator.DIV, false, false)},
+		
+		{"\\(%number%\\)[ ]^[ ]\\(%number%\\)", new PatternInfo(Operator.EXP, true, true)},
+		{"\\(%number%\\)[ ]^[ ]%number%", new PatternInfo(Operator.EXP, true, false)},
+		{"%number%[ ]^[ ]\\(%number%\\)", new PatternInfo(Operator.EXP, false, true)},
+		{"%number%[ ]^[ ]%number%", new PatternInfo(Operator.EXP, false, false)},
+		
+	});
+	
+	static {
+		Skript.registerExpression(ExprArithmetic.class, Number.class, ExpressionType.PATTERN_MATCHES_EVERYTHING, patterns.getPatterns());
+	}
+	
+	@SuppressWarnings("null")
+	private Expression<? extends Number> first;
+	@SuppressWarnings("null")
+	private Expression<? extends Number> second;
+	@SuppressWarnings("null")
+	private Operator op;
+	
+	@SuppressWarnings("null")
+	private Class<? extends Number> returnType;
+	private boolean integer;
+	
+	// A chain of expressions and operators, alternating between the two. Always starts and ends with an expression.
+	private final List<Object> chain = new ArrayList<>();
+	
+	// A parsed chain, like a tree
+	private ArithmeticGettable arithmeticGettable;
+	
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+		first = (Expression<? extends Number>) exprs[0];
+		second = (Expression<? extends Number>) exprs[1];
+		
+		PatternInfo patternInfo = patterns.getInfo(matchedPattern);
+		op = patternInfo.operator;
+		
+		if (op == Operator.DIV || op == Operator.EXP) {
+			returnType = Double.class;
```

Comment:
Not all divisions/exponential operations end up being real numbers, but since Skript converts the type when needed, this might be just fine like it is now. Take this line into account if any conversion problems appear after merge, though.

Reply:
I took this part from the old expression, so it should be fine.